arviz/stats/density_utils.py

```diff
@@ -946,20 +947,37 @@ def get_bins(values):
     It is considered a robusts version of the Scott rule as the IQR is less affected by outliers
     than the standard deviation. However, the IQR depends on fewer points than the standard
     deviation, so it is less accurate, especially for long tailed distributions.
+
+    -------
+    Adding functionality for non integer data
+
+    Segment only works if input dtype is not int32. Width of bins is computed with
+    numpy.histogram_bin_edges with bins='auto'. Auto mode uses the maximum of Sturges and
+    Freedman-Diaconis estimators to provide bin margins. Rest of the parameters to
+    numpy.histogram_bin_edges are default.
+
+
     """
-    x_min = values.min().astype(int)
-    x_max = values.max().astype(int)
 
-    # Sturges histogram bin estimator
-    bins_sturges = (x_max - x_min) / (np.log2(values.size) + 1)
+    if values.dtype == "int32":
```

Comment:
we should use `"i"` and/or `"f"`, more often than not data will be `int64` instead of `int32`, but the code should work in all cases. See for example https://github.com/arviz-devs/arviz/blob/main/arviz/plots/backends/matplotlib/ppcplot.py#L130

Reply:
Ofcourse, thanks for the example, using same technique now